@navikt/core/react/src/modal/Modal.tsx

```diff
@@ -79,111 +89,134 @@ type ModalLifecycle = {
+    if (useContext(ModalContext)) console.error("Modals should not be nested");
 
-    const onModalCloseRequest = (e) => {
-      if (shouldCloseOnOverlayClick || e.type === "keydown") {
-        onClose();
-      } else if (buttonRef.current) {
-        buttonRef.current.focus();
+    useEffect(() => {
+      if (needPolyfill && modalRef.current) {
+        dialogPolyfill.registerDialog(modalRef.current);
       }
-    };
+    }, [modalRef]);
 
-    const getParentSelector = () => {
-      if (parentSelector) {
-        return parentSelector;
+    useEffect(() => {
+      // We need to have this in a useEffect so that the content renders before the modal is displayed,
+      // and in case `open` is true initially.
+      if (modalRef.current && open !== undefined) {
+        if (open && !modalRef.current.open) {
+          modalRef.current.showModal();
+        } else if (!open && modalRef.current.open) {
+          modalRef.current.close();
+        }
       }
-      return rootElement !== undefined
-        ? () => rootElement as HTMLElement
-        : undefined;
-    };
+    }, [modalRef, open]);
 
     return (
-      <ReactModal
-        {...rest}
-        parentSelector={getParentSelector()}
-        style={style}
-        isOpen={open}
+      <dialog
         ref={mergedRef}
-        className={cl("navds-modal", className)}
-        overlayClassName={cl("navds-modal__overlay", overlayClassName)}
-        shouldCloseOnOverlayClick={shouldCloseOnOverlayClick}
-        shouldCloseOnEsc={shouldCloseOnEsc}
-        onRequestClose={(e) => onModalCloseRequest(e)}
-        aria={{
-          describedby: ariaDescribedBy,
-          labelledby: ariaLabelledBy,
-          modal: ariaModal,
+        className={cl("navds-modal", className, {
+          "navds-modal--polyfilled": needPolyfill,
+        })}
+        onCancel={(event) => {
+          // FYI: onCancel fires when you press Esc
+          if (onBeforeClose && onBeforeClose() === false) {
+            event.preventDefault();
+          } else if (onCancel) onCancel(event);
         }}
-        contentLabel={contentLabel}
+        aria-labelledby={
+          ariaLabelledBy ?? (header ? internalAriaId : undefined)
+        }
+        {...rest}
       >
-        {children}
-        {closeButton && (
-          <Button
-            className={cl("navds-modal__button", {
-              "navds-modal__button--shake": shouldCloseOnOverlayClick,
-            })}
-            size="small"
-            variant="tertiary-neutral"
-            ref={buttonRef}
-            onClick={onClose}
-            icon={<XMarkIcon title="Lukk modalvindu" />}
-          />
-        )}
-      </ReactModal>
+        <ModalContext.Provider
+          value={{
+            closeHandler: getCloseHandler(modalRef, header, onBeforeClose),
+          }}
+        >
+          {header && (
+            <ModalHeader>
+              {header.label && (
+                <Detail
+                  as="label"
```

Comment:
Er det noen fordeler ved bruk av `label` vs p/span her?

Reply:
Godt spørsmål. Tenkte kanskje `<label>` var hakket mer semantisk her. Har lest at det er greit å bruke det utenom inputs også. Men har hørt med Vegard nå, som syntes det var mer passende med `<p>`, så da går jeg for det.